consensus/consortium/v2/finality/consortium_header.go

```diff
@@ -305,10 +307,16 @@ func (extraData *HeaderExtraData) EncodeRLP() ([]byte, error) {
 		cp[i] = validatorWithBlsPubRLP{
 			Address:      val.Address,
 			BlsPublicKey: val.BlsPublicKey.Marshal(),
+			StakedAmount: val.StakedAmount,
 		}
 	}
 	ext.CheckpointValidators = cp
 
+	if extraData.BlockProducers != nil {
+		ext.BlockProducers = make([]common.Address, len(extraData.BlockProducers))
+		copy(ext.BlockProducers, extraData.BlockProducers)
```

Comment:
I think we don't need deep copy here, isn't `ext.BlockProducers = extraData.BlockProducers` necessary here?